core/store/models/run_result.go

```diff
@@ -0,0 +1,42 @@
+package models
+
+import (
+	"fmt"
+
+	"github.com/tidwall/gjson"
+	null "gopkg.in/guregu/null.v3"
+)
+
+// RunResult keeps track of the outcome of a TaskRun or JobRun. It stores the
+// Data and ErrorMessage, and contains a field to track the status.
+type RunResult struct {
+	ID           uint        `json:"-" gorm:"primary_key;auto_increment"`
+	Data         JSON        `json:"data" gorm:"type:text"`
+	ErrorMessage null.String `json:"error"`
+}
```

Comment:
The names are beginning to feel confusing.  As there is RunInput, RunOutput, and RunResult, what is the difference between a RunOutput and a RunResult?  It hints to me two concepts are being called a Run.  Is the interpretation correct?

Reply:
Agreed. Seems like they are conceptually one thing, why keep different models?

Reply:
Strong disagree, the differences are substantial, an example: if you merge RunOutput on to RunResult, then which status do you check on a run? Run.status? or run.result.status?